### ISSUE TITLE
Facebook for WooCommerce - API key is no longer valid

### ISSUE THREAD

**Issue by nimbijewelry** [facebookincubator/facebook-for-woocommerce#591]
Hello!

On my Wordpress site (wwwi.nimbi.at) I us the Plugin "facebook for WooCommerce".
Sincce now, there is is error message: "**Your API key is no longer valid. Please click "Settings > Advanced Options > Update Token".**"

I tryed everything I know to fix this, but I haven't found the solution so far. 

It would be great if anyone could help my to fix this problem.

THANk YOU!!

Ingrid

**Comment by mengyingdu** [comment]
## Automated Support Message

Hello,

Thanks for your interest in our plugin!

If your issue is a feature request or feedback for the developers, reply with `Feature Request` or `Feedback`.
If you are a developer or have a question for the developers related to code, reply with `Developer Question`.
If you are running ads with us, include `Advertiser` in your reply.

If you have encountered a bug or a problem with the plugin, the steps below contain information to help you resolve the issue yourself and provide us information we need to help you with your problem.

## Search Existing Issues

**First, please [search our issues](https://github.com/facebookincubator/facebook-for-woocommerce/issues?q=) to see if your problem has been reported before.**
There may already be a workaround or a solution to your problem. You can search our issues using [the search bar near the top of our issues page](https://github.com/facebookincubator/facebook-for-woocommerce/issues?q=); be sure to search both _open_ and _closed_ issues.
If you do find a similar issue, leave a comment on that issue and close this one. Your comment helps us prioritize which existing issues are important and urgent.

## Common Resolution Methods

**If you are experiencing a problem or bug and have not found any similar issue, please try the following:**

* Upgrade to the latest version of the plugin.
* Try to `Resync` products using the button on the plugin config page
* Reset the plugin by selecting `Reconfigure > Advanced Options > Delete`, and setting up this plugin again.
* Try to deactivate other plugins that you think may be interfering, then reset and set up again.

## Necessary Information

**If the above steps did not help, please provide the following information in your next comment in order for us to debug your problem:**

* WooCommerce version
* WordPress version
* Plugin version (please note: we only support up-to-date versions of the plugin)
* A screenshot of where you see the problem (for example: if it is on Facebook, a screenshot of Facebook, if it is on WooCommerce, a screenshot of WooCommerce. If it is in multiple places, multiple screenshots.)
* The URL of your website
* The URL of your page
* The URL(s) of where you took your screenshot(s)
* (If applicable) Detailed steps you took in order to see the problem

Please include `Debug Ready` in your reply so our automated system can flag your issue for manual review. Please note that due to a high volume of issues, we may not be able to respond immediately.

Thanks for your patience,
The Facebook Dev Team (@mengyingdu, @dmitridr, @sammul, @jieminz22)

**Comment by jieminz22** [comment]
## Automated Support Message - Issue Closed

Hello,

Because it has been 30 days since this issue was updated and you have not included `Debug Ready` in your reply, our automated system will now close your issue.
If you are still having this problem, please open a new issue and reference this one.

For more information on our response times and support policy, visit [Facebook Support on GitHub](https://github.com/facebookincubator/facebook-for-woocommerce/wiki/Facebook-Support-on-GitHub).

Thank you,
The Facebook Dev Team

**Comment by VijayendraM** [comment]
Debug Ready

Facebook for WooCommerce : Version 1.9.11
WooCommerce : Version 3.5.7
WordPress:  version 5.1.1
Screenshot -  ![FB_Error](https://user-images.githubusercontent.com/15442787/54887601-3783f800-4e95-11e9-87fa-cb14b667e326.jpg)

The URL of your website : www.YouAndBeauty.com
The URL of your page : https://www.facebook.com/Shop.YouAndBeauty/

What did I do ? 
I had to change the currency and the name of the ads account. As per the FB Help article, I created a new one (see screenshot) with the name as "You and Beauty" and it asked to deactivate the old one "Vijayendra Meshram"
![image](https://user-images.githubusercontent.com/15442787/54887636-bd07a800-4e95-11e9-86fb-d96a14935ded.png)

For troubleshooting I uninstalled and reinstalled the plugin but issue persists.
When I click on Settings it does not show me my existing page but lets me create a new page
Tried creating a new page "Testing" and only that shows up. The original page "You and Beauty" does not show up in the list. 

![image](https://user-images.githubusercontent.com/15442787/54887695-51720a80-4e96-11e9-9b54-cbf6d0a03372.png)

Regards, 
Vijayendra

**Comment by VijayendraM** [comment]
Fixed it : Unpublish the page and then re-publish
https://user-images.githubusercontent.com/15442787/54887939-9b5bf000-4e98-11e9-93bc-609af0813c1e.png

Once this is done
Settings in the FB for woocommerce plugin : now the page is there.
Select the option "update token" and all looks good. :)

Regards,
Vijayendra

**Comment by disenosanpedro** [comment]
Hello .. 2 days ago the plugin worked smooth .. Then i switched / changed the FB page to other .. since yesterday i am not able to connect woocommerce with the facebook ... it says: Your API key is no longer valid. Please click "Settings > Advanced Options > Update Token".   There is no reset or reconfigure .. Im using the last version of Woocommerce and last version of Facebook to Woocommerce.  I have tried all the plugins .. with pixel inside, without , downloaded the one from facebook there's no way to get rid of this API issue .. and when i click on Settings inside the plugin i get from facebook:  

Sorry, something went wrong.
We're working on getting this fixed as soon as we can.

Anyone have any idea how to fix this please? Or what do i have to do ? 

Best Regards

**Comment by mrkapable007** [comment]
This didnt work for me , i had to delete this plugin . 

anyfix for this ?

**Comment by mrkapable007** [comment]
I unpublished and published , but still didnt work . what is the fix for this ?

**Comment by YoChes** [comment]
I have the same problem. Anyone have an idea what's going on?

**Comment by suneth2** [comment]
same problem here, i tried few times and update the key but no luck

**Comment by Precies** [comment]
Same!  also did the unpublished and published thing. Didn't work for me...

**Comment by arzu-poolie** [comment]
Hi everyone, I have the same issue - already tried some of the tricks online but nothing works for me. Is there going to be an update of the plugin? Did someone announce a date where the new update is going to be released? This bug concerns a lot of users as I've seen in many github threads. 

Does anybody have a solution yet or even a workaround until the issue is being fixed by FB?

**Comment by babynosoucy** [comment]
Hello,
Même problème chez moi- j'espère une mise à jour positive rapide !

**Comment by tangquangminh677** [comment]
Xin chào mọi người, tôi có cùng một vấn đề - đã thử một số thủ thuật trực tuyến nhưng không có gì hiệu quả với tôi. Liệu sẽ có một bản cập nhật của plugin? Có ai đó đã thông báo một ngày mà bản cập nhật mới sẽ được phát hành? Lỗi này liên quan đến rất nhiều người dùng như tôi đã thấy trong nhiều luồng github.

Có ai có giải pháp nào hoặc thậm chí là một cách giải quyết cho đến khi vấn đề được FB khắc phục?
https://shopsosanh.vn/
